### PR TITLE
Add ADR about carrying applications over to the next cycle

### PR DESCRIPTION
**Pull request by duncanjbrown**
## Context

Writing up decisions following a meeting on 12th August about #2681.

## Changes proposed in this pull request

Add an ADR about what we decided.

## Guidance to review

- Does this fairly reflect the meeting?
- Did we miss anything?

## Link to Trello card

N/A

## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: adr/0015-carrying-over-applications.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  cloning the ApplicationForm and removing the courses
+- We will copy references onto the carried-over application, but only if
+  they’re complete. Referees who had been contacted but had not responded
+  before the application was carried over will need to be cancelled.
+- The applications that were carried over will remain in the database without
+  any further state change
+- Applications which were not yet sent to the provider at end of cycle and also
+  not carried over will still be in the database — we would like to mark these
+  with a new state equivalent to “incomplete at end of cycle”. This state would
+  never be visible to providers.
```
Comment (tijmenb) [inline adr/0015-carrying-over-applications.md:79]: What do we do with applications in awaiting references state? Do we want to cancel these references?

Comment (duncanjbrown) [reply adr/0015-carrying-over-applications.md:79]: See two bullet points up (unless I've missed something)

File: adr/0015-carrying-over-applications.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+#### Before the application reaches the provider
+
+- Withdrawn
+- Unsubmitted
+- Ready to send to provider
+
+#### After the application can’t progress any further
+
+- Conditions met (recruited)
```
Comment (mnacos) [inline adr/0015-carrying-over-applications.md:27]: `Conditions met (recruited)` appears both here and in the 'does not make sense' section. Which one should stay?

File: adr/0015-carrying-over-applications.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+### Carrying over an application does not make sense in the following states
+
+#### While the application is already under consideration by the provider
+
+- Awaiting provider decision
+
+#### When the application already has an offer in flight
+
+- Offer
+- Meeting conditions
```
Comment (mnacos) [inline adr/0015-carrying-over-applications.md:42]: What is 'Meeting conditions'? Is it a future state (some conditions have been met while others are pending)?

File: adr/0015-carrying-over-applications.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+will be some candidates who could benefit from their application being carried
+over to the next cycle. Carrying over an application means the candidate can
+apply to courses in the new recruitment cycle without having to fill in the
+whole application form again.
+
+### Carrying over an application makes sense in the following states
+
+#### Before the application reaches the provider
+
+- Withdrawn
```
Comment (mnacos) [inline adr/0015-carrying-over-applications.md:21]: Is this so that the candidate can un-withdraw and apply again?

**Comment by duncanjbrown** [comment]
Thanks for your comments @mnacos! I've removed "Conditions met" from the "makes sense to carry over" section and added some additional text explaining the rationale for when things are carry-over-able.

**Comment by duncanjbrown** [comment]
re application choices being in mixed states, I think the rule of thumb is that if at least one choice is not yet "finished" it is not possible to carry over. Once they're all done for, it is.

File: adr/0015-carrying-over-applications.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+- Offer
+- Meeting conditions (i.e. offer accepted)
+- Conditions met (recruited)
+
+### Copying the Apply again approach
+
+The current approach for moving applications into Apply again is to copy the
+entire application (including references) and invite the user to add a new
+course choice. This approach seems like it will work here too, with a couple of
```
Comment (mnacos) [inline adr/0015-carrying-over-applications.md:55]: This is a great addition, thanks!

**Review by mnacos — APPROVED** [review]
Explaining the reasoning behind which states will are carried over makes this document clearer :rocket:

**Pull request merged**